roles/fio_distributed/templates/server_vm.yml.j2

```diff
@@ -13,52 +13,53 @@ spec:
       cores: {{ workload_args.vm_cores | default(1) }}
     devices:
       disks:
-        - name: registrydisk
-        - name: cloudinitdisk
+        - disk:
+            bus: virtio
+          name: registrydisk
+        - disk:
+            bus: virtio
+          name: cloudinitdisk
 {% if workload_args.storageclass is defined
     or workload_args.hostpath is defined %}
         - disk:
             bus: virtio
           name: data-volume
           serial: data
+{% else %}
+        - disk:
+            bus: virtio
+          name: emptydisk
 {% endif %}
     resources:
       requests:
         memory: {{ workload_args.vm_memory | default('5G') }}
   volumes:
     - name: registrydisk
       containerDisk:
-        image: {{ workload_args.vm_image | default('kubevirt/fedora-cloud-container-disk-demo:latest') }}
+        image: {{ workload_args.vm_image | default('quay.io/kubevirt/fedora-cloud-container-disk-demo:latest') }}
     - name: cloudinitdisk
       cloudInitNoCloud:
         userData: |-
           #cloud-config
-          password: fedora
+          password: ripsaw
           chpasswd: { expire: False }
           bootcmd:
+            - "mkdir -p {{ fio_path }} || true"
 {% if workload_args.storageclass is defined
     or workload_args.hostpath is defined %}
-            - "mkdir -p {{ fio_path }} || true"
             - mkfs.ext4 /dev/disk/by-id/virtio-data
             - "mount /dev/disk/by-id/virtio-data {{ fio_path }}"
+{% else %}
+            - mkfs.ext4 /dev/vdc
+            - "mount /dev/vdc {{ fio_path }}"
 {% endif %}
           runcmd:
-{% if workload_args.vm_image is undefined %}
-            - dnf install -y fio wget curl python3 python3-pip
-{% endif %}
-            - cd /tmp
-{% if workload_args.metadata is defined
-    and (workload_args.metadata.collection|default(false))
-    and (workload_args.metadata.targeted|default(true)) %}
-            - export ES_SERVER={{ elasticsearch.url }}
-{% if workload_args.vm_image is undefined %}
-            - wget https://raw.githubusercontent.com/cloud-bulldozer/bohica/master/stockpile-wrapper/stockpile-wrapper.py
-            - pip3 install elasticsearch-dsl openshift kubernetes redis
-{% endif %}
-            # TODO: start stockpile-wrapper script
-            python3 stockpile-wrapper.py -s {{ elasticsearch.url }} -u {{ uuid }} -n "$my_node_name" -N "$my_pod_name" --redisip {{ bo.resources[0].status.podIP }} --redisport 6379 --force
-{% endif %}
-            - fio --server
+            - dnf install -y podman
+            - "img=`podman create {{ workload_args.image | default('quay.io/cloud-bulldozer/fio:latest') }}`"
+            - fs=`podman mount $img`
+            - "mkdir -p $fs/{{ fio_path }} || true"
+            - "mount -o bind /{{ fio_path }} $fs/{{ fio_path }}"
+            - "chroot $fs bash -c 'cd /{{ fio_path }}; fio --server'"
```

Comment:
where does port 8765 (fio --server port) get opened up?  Or is the CNV VM just not using a firewall?

Reply:
This is a very interesting point I did not think about, because the Kubevirt-provided containerDisk images use "cloud editions" that do not install a firewall (for ex. mentioned in fedora [docs](https://docs.fedoraproject.org/en-US/fedora-server/virtualization-vmcloud/)). But longer term we should consider how to extend this to *any* type of guest image and would need to handle the various firewall cmds in that case..

Reply:
I know it has been a while but adding my 2c:
Rather than mounting the containers root to the host, then bind-mounting the device, I would forward the device into the container with the `podman run --volume {{ fio_path}}:/target ` option.

Then I would expect that the entrypoint of the fio container would call `fio --server` anyways. That for me is a more straightforward way that future people will know and understand and we don't need the chroot.

With that, runcmd would shrink to:

```
            - dnf install -y podman
            - "podman run --volume {{ fio_path}}:/target --entrypoint 'fio --server --directory=/target' {{ workload_args.image | default('quay.io/cloud-bulldozer/fio:latest') }}"
```
(untested)

Reply:
@mulbc looks simpler and more like typical podman/docker usage to me, but maybe I'm missing something about why Jen did it the other way?  @jeniferh ?

Reply:
@mulbc thanks for the review. I wanted to avoid testing inside a container environment if possible since we know it could add some overhead, we could probably carefully tune around that by making the container privileged, turning off seccomp, enabling all CAPs etc. but the idea was chroot is simply a way to use what the container image provides without actually adding another layer to the data-path by running from w/in a full container.